### DISCUSSION TITLE
Improve documentation on Choose action

### DISCUSSION THREAD

**Discussion by SimonGAndrews** [statelyai/xstate#1312] [General]
I struggled to implement the chose action functions.   I couldnt find an example of the use of Choose Action on the Github and the description in the guides took a while with experiments to understand.  Esentially  tt was not initially clear to me that the a cond was one and the same as a condition defined in a guard.    Guards are disucssed within the Guarded transactions section and its not obvious they are needed to suppot other functionality (eg choose action)

Suggestions:
1) give guards / conditions thier own section in the  guides so they can be referred to in guarded transitions and choose action sections.   in the guards section state that they are used for choose action.
 
In the choose Actions section of the guides:

2) - After the existing: 
Each "conditional actions" object in cond has these properties:
    * actions - the action objects to execute
    * cond? - the condition for executing those actions
  put a few words to make the link to guard/condition definitions
 eg   Conditions are named guard conditions as defined in the guard options of the machine (link) or are defined in line (link) 

3) Include in the example in choose action the definion of the guard that defines the conditions in the example.  To help see the dependency on guard/condition definition.  eg 
 guards: {
        cond1: (context, event, { cond }) => {
            return ( false)   // setting to true results in cond1 actions being executed
            },
        cond2: (context, event, { cond }) => {
            return ( true)
            },
     }

Sorry to be picky , but this may hlep others.

**Comment by harshagarwalnyu** [comment]
The confusion is very understandable — the XState docs treat `choose` and `guards` in separate sections without making the dependency explicit. Here's the missing link:

**`choose` requires guards — they are the same thing.**

A `choose` action selects the first matching condition and runs its actions. Those conditions ARE guards, defined in the `guards` object of your machine config:

```js
import { createMachine, assign, choose } from 'xstate';

const machine = createMachine(
  {
    context: { count: 0, role: 'user' },
    on: {
      INCREMENT: {
        actions: 'incrementWithChoose',
      },
    },
  },
  {
    actions: {
      incrementWithChoose: choose([
        {
          // This `cond` references a named guard defined below
          cond: 'isAdmin',
          actions: assign({ count: (ctx) => ctx.count + 10 }),
        },
        {
          // Inline guard (no name needed)
          cond: (context) => context.count < 5,
          actions: assign({ count: (ctx) => ctx.count + 2 }),
        },
        {
          // No cond = default (always matches if reached)
          actions: assign({ count: (ctx) => ctx.count + 1 }),
        },
      ]),
    },
    guards: {
      // Named guard — referenced by string in choose's `cond`
      isAdmin: (context) => context.role === 'admin',
    },
  }
);
```

**Key rules:**
1. `cond: 'guardName'` looks up `guards.guardName` in machine options
2. `cond: (ctx, event) => boolean` is an inline guard — no name needed
3. Conditions evaluate top-to-bottom — first truthy `cond` wins
4. No `cond` = always true (default case — put it last)

**The docs gap you identified:** guards are introduced under "Guarded Transitions" as a transition feature, but `choose` also uses them as action conditions. They're the same mechanism in two different contexts. A side-by-side example showing `choose` + `guards` together would fix this confusion for everyone.
